### ISSUE TITLE
[Bug]: Scanning external app QR code does not switch to the proper network for the QR

### ISSUE THREAD

**Issue by chrisleewilcox** [MetaMask/metamask-mobile#12081]
### Describe the bug

When I scan a Coinbase QR code for the Ethereum Mainnet using mobile app and mobile app does not have Ethereum Mainnet selected the mobile app does not switch to the correct network. 

### Expected behavior

Mobile app should switch the the proper network for the QR being scanned.

### Screenshots/Recordings

Mobile app is not switch to Ethereum Mainnet after scanning this CB QR from the CB mobile app...
![IMG_2867](https://github.com/user-attachments/assets/a9043ebd-f94b-438e-b305-168585cd97dd)

### Steps to reproduce

GIVEN I Linea is selected for mobile app
WHEN I scan a Coinbase QR for Ethereum Mainnet using mobile app
THEN mobile app switches to Ethereum Mainnet (FAILED: network is not switched)
AND correct address from QR is populated for To field in Send flow

### Detection stage

During release testing

### Version

7.34.0

### Build type

None

### Device

Any

### Operating system

iOS, Android

**Comment by Unik0rnMaggie** [comment]
Reproduced on RC version 7.36.0 build from here, for the following examples:

Current selected network is Linea, and when scanning the QR code for all the below networks, the current selected network does not change in the MetaMask app.

Examples:

        | NETWORK  | ADDRESS_QR_CODE_LINK                                 | ADDRESS                                    
        
        | Ethereum  | https://www.screencast.com/t/EDOnQunzTbci    | 0xcDb405e0bFA1554F6668CF55d3D7e04afbEe0708 
        | Avalanche | https://www.screencast.com/t/Ocy7Gunp4X       | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
        | Polygon     | https://www.screencast.com/t/uusyaN9Agn        | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
        | Binance     | https://www.screencast.com/t/MM37Co8UERAv | 0xe49267bCDdd5E137Ea83B24731491b3D8C4B5c55 
        
        
        

https://github.com/user-attachments/assets/f4e1aff4-b755-42eb-bb0e-e00ae463c668

**Comment by Unik0rnMaggie** [comment]
Present in [v7.37.0 (1511) RC](https://app.bitrise.io/app/be69d4368ee7e86d/installable-artifacts/6b30aa4ce517530e/public-install-page/c143a7405ced13a8ca26428b1e8fe980)

Current selected network is Linea, and when scanning the QR code for the below addresses, the current selected network does not change in the MetaMask app (remains on Linea)

Examples:

    | NETWORK  | ADDRESS_QR_CODE_LINK                                 | ADDRESS                                    
    
    | Ethereum  | https://www.screencast.com/t/EDOnQunzTbci    | 0xcDb405e0bFA1554F6668CF55d3D7e04afbEe0708 
    | Avalanche | https://www.screencast.com/t/Ocy7Gunp4X       | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
    | Polygon     | https://www.screencast.com/t/uusyaN9Agn        | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
    | Binance     | https://www.screencast.com/t/MM37Co8UERAv | 0xe49267bCDdd5E137Ea83B24731491b3D8C4B5c55 
    
    
    

https://github.com/user-attachments/assets/7dfc0f63-ecab-455a-b26e-fff08fd344ee

**Comment by pnarayanaswamy** [comment]
Same issue reproduced on v7.38.0

**Comment by Unik0rnMaggie** [comment]
Present in v7.39.0. with the same behavior described above:

for the below examples, when scanning the QR code, the enabled network in MetaMask does not change


| NETWORK  | ADDRESS_QR_CODE_LINK                                 | ADDRESS                                    

| Ethereum  | https://www.screencast.com/t/EDOnQunzTbci    | 0xcDb405e0bFA1554F6668CF55d3D7e04afbEe0708 
| Avalanche | https://www.screencast.com/t/Ocy7Gunp4X       | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
| Polygon     | https://www.screencast.com/t/uusyaN9Agn        | 0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121 
| Binance     | https://www.screencast.com/t/MM37Co8UERAv | 0xe49267bCDdd5E137Ea83B24731491b3D8C4B5c55

**Comment by Unik0rnMaggie** [comment]
Present in v7.40.0 (1540)

**Comment by AlexHerman1** [comment]
reproduced today on v7.46.2

**Comment by AlexHerman1** [comment]
updating to sev1 as this could easily lead to user funds loss

**Comment by wenfix** [comment]
The problem is occurring at the QR code level. The QR codes shown in the examples don't include a `chain_id` parameter as required by [ERC-681](https://eips.ethereum.org/EIPS/eip-681). 

Here's the current QR code content:
- Example 1 (Coinbase)
<img width="1240" height="375" alt="Image" src="https://github.com/user-attachments/assets/8c15cd5b-c19b-4106-baeb-82128e32a63c" />

- Example 2 ([Avalanche](https://github.com/MetaMask/metamask-mobile/issues/12081#issuecomment-2503280390)):
<img width="1239" height="372" alt="Image" src="https://github.com/user-attachments/assets/d0a03c1d-66a1-4fe7-82eb-0418e2c3e7b5" /> 
 
According to ERC-681, the QR code content should follow this format:

```javascript
ethereum:0x79821Ea7aB5c5a34A24b2fd547c544ac15a7b121@1
``` 

Where the `chain_id` (in this case "1" for Ethereum mainnet) appears after the wallet address, separated by an "@" symbol.
Without this chain identifier in the QR code content, the parser cannot determine which network to use for the transaction.

I've debugged the QR content after it is parsed, which happens [here](https://github.com/MetaMask/metamask-mobile/blob/7da38e7bf8499988fd7ebcbeb1e04522ad847278/app/components/Views/QRScanner/index.tsx#L222), and it yields the same results as the screenshots. 
When I manually modified the parsed content string to include a `chain_id` parameter, the app correctly triggered a chain switch.


TL;DR: The app is behaving as expected, the QR codes are missing the required parameter in order to trigger a chain switch.

CC: @adonesky1

**Comment by adonesky1** [comment]
We will report the issue to Coinbase and close the ticket on our side.
